### PR TITLE
List AWS roles and codesmells

### PR DESCRIPTION
**Pull request by nithyal1**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/SelfSupportController.java [hunk 1]
```diff
@@ -483,4 +483,17 @@
...
 		return selfSupportService.transferSafe(token, safeTransferRequest, userDetails);
 	}
+	
+	/**
+	 * To list aws ec2 roles
+	 * @param request
+	 * @param token
+	 * @return
+	 */
+	@ApiOperation(value = "${AWSAuthControllerV2.listRoles.value}", notes = "${AWSAuthControllerV2.listRoles.notes}")
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/SelfSupportController.java:493]: This should be SelfSupportController.listRoles.value

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SelfSupportService.java [hunk 2]
```diff
@@ -1219,4 +1220,36 @@ else if (matchingPolicies.stream().anyMatch(p-> p.equals("r"+itemName)) || match
...
+			      put(LogMessage.MESSAGE, "Trying to get list of AWS roles").
+			      put(LogMessage.APIURL, ThreadLocalContext.getCurrentMap().get(LogMessage.APIURL).toString()).
+			      build()));
+		String path = TVaultConstants.AWS_USERS_METADATA_MOUNT_PATH + "/" + userDetails.getUsername();
+		Response response = new Response();
+		if (userDetails.isAdmin()) {
+			return awsAuthService.listRoles(token);
+		}
+		else {
+			response = reqProcessor.process("/auth/aws/rolesbyuser/list",PATHSTR+path+"\"}",userDetails.getSelfSupportToken());
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SelfSupportService.java:1243]: Need to use user token here for normal user

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AppRoleService.java [hunk 3]
```diff
@@ -1353,7 +1331,7 @@ else if(response.getStatusCode().equals(HttpStatus.OK)) {
 		boolean canAddAppRole = ControllerUtil.canAddPermission(path, token);
 		if(canAddAppRole){
 
-			log.info("Associate approle to SDB -  path :" + path + "valid" );
+			log.info("Associate approle to SDB -  path : {} valid", path );
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AppRoleService.java:1334]: Can we use the required log format

**Review by sarathmohan4 — CHANGES_REQUESTED** [review]

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/SelfSupportController.java (hunk 1 above)
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/SelfSupportController.java]: changed to SelfSupportController.listRoles.value

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AppRoleService.java (hunk 3 above)
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AppRoleService.java]: changed to the required log format

**Review by sarathmohan4 — APPROVED** [review]

**Review by alexrajan3 — APPROVED** [review]

**Pull request merged**